utils/build-script-impl

```diff
@@ -2265,17 +2440,17 @@ for deployment_target in "${HOST_TARGET}" "${CROSS_COMPILE_TOOLS_DEPLOYMENT_TARG
                 if [[ -z "${INSTALL_LLDB}" ]] ; then
                     continue
                 fi
-                case "$(uname -s)" in
-                    Linux)
+                case ${host} in
+                    linux-*)
                         ;;
-                    FreeBSD)
+                    freebsd-*)
                         ;;
-                    CYGWIN_NT-10.0)
+                    cygwin-*)
                         ;;
-                    Darwin)
+                    macosx-*)
                         set_lldb_build_mode
                         pushd ${LLDB_SOURCE_DIR}
-                        xcodebuild -target toolchain -configuration ${LLDB_BUILD_MODE} install ${lldb_xcodebuild_options[@]} DSTROOT="${INSTALL_DESTDIR}" LLDB_TOOLCHAIN_PREFIX="${TOOLCHAIN_PREFIX}"
+                        xcodebuild -target toolchain -configuration ${LLDB_BUILD_MODE} install ${lldb_xcodebuild_options[@]} DSTROOT="${host_install_destdir}"
```

Comment:
There would only be a problem if you're on OSX, building a compiler for a linux host, and you wanted that linux host to have some other install prefix than `/usr`. Then the OSX toolchain that gets placed in the install location wouldn't have the correct prefix.

For non-cross-compilers, the changes would be for OSX: you would always specify a `/usr` install prefix, strip the toolchain prefix from any scripts. The products will still be in the same location as before, just the parameter changes.

Alternatively, we could hard-code the `/usr` install prefix for Darwin toolchains and ignore the `INSTALL_PREFIX` parameter.